giskard-server/src/main/java/ai/giskard/service/InitService.java

```diff
@@ -292,12 +324,17 @@ private void uploadDataframe(String projectKey) {
         }
     }
 
-    private void uploadModel(String projectKey) {
-        Resource modelResource = resourceLoader.getResource("classpath:demo_projects/" + projectKey + "/model.pkl.zst");
-        Resource requirementsResource = resourceLoader.getResource("classpath:demo_projects/" + projectKey + "/requirements.txt");
+    private void uploadModel(String projectKey, String filename) {
+        String pathToModel = CLASSPATH + PROJECTDIR + projectKey + "/models/" + filename + ".model.pkl.zst";
+        String pathToRequirements = CLASSPATH + PROJECTDIR + projectKey + "/requirements/" + filename + ".requirements.txt";
+        Resource modelResource = resourceLoader.getResource(pathToModel);
+        Resource requirementsResource = resourceLoader.getResource(pathToRequirements);
+        ModelUploadParamsDTO modelDTO = projects.get(projectKey).modelParams;
+        modelDTO.setName(modelDTO.getName() + " " + filename);
```

Comment:
you're modifying the shared `ModelUploadParamsDTO`. This results in model names like "German credit score 0 1" because on the first iteration "0" was already appended to the name. We should create a copy of `projects.get(projectKey).modelParams` here before modifying it.